backend/app/api/deps.py

```
@@ -10,21 +10,32 @@
-    if not user:
-        raise HTTPException(status_code=status.HTTP_401_UNAUTHORIZED, detail="Invalid credentials")
-    return user
+    try:
+        payload = decode_access_token(token)
+        user_id = payload.get("sub")
+        if not user_id:
+            raise HTTPException(status_code=401, detail="Invalid token")
+        
+        query = select(Users).where(
+            Users.user_id == int(user_id),
+            Users.is_active == True,
```

Ruff (0.12.2)

Avoid equality comparisons to `True`; use `Users.is_active:` for truth checks

Replace with `Users.is_active`

(E712)